File: backend/Entities/Travel/Travel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Entities.Travels
{

    [Table("travels")]
    public class Travel
    {
        [Key]
        [Column("pk_travel_id")]
        public int TravelId { get; set; }

        [Required]
        [MaxLength(255)]
        [Column("travel_name")]
        public string? TravelName { get; set; }

        [Required]
        [MaxLength(255)]
        [Column("destination")]
        public string? Destination { get; set; }

        [Column("purpose")]
        public string? Purpose { get; set; }

        [Required]
        [Column("start_date")]
        public DateTime StartDate { get; set; }

        [Required]
        [Column("end_date")]
        public DateTime EndDate { get; set; }

        [Required]
        [Column("fk_created_by")]
        public int CreatedBy { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        [ForeignKey("CreatedBy")]
        public virtual User? Creator { get; set; }

        public ICollection<TravelAssignment> Assignments { get; set; } = new List<TravelAssignment>();

        public ICollection<TravelDocument> Documents { get; set; } = new List<TravelDocument>();

        public ICollection<Expense> Expenses { get; set; } = new List<Expense>();

    }

}

File: frontend/src/types/travel.ts
export type TravelAssigned = {
  travelId: number
  travelName: string
  destination: string
  purpose?: string
  startDate: string
  endDate: string
  assignedEmployeeIds?: number[]
}

export type TravelAssignment = {
  assignId: number
  travelId: number
  travelName: string
  destination?: string
  purpose?: string
  startDate: string
  endDate: string
  assignedEmployeeIds?: number[]
}

export type TravelDetail = {
  travelId: number
  travelName: string
  destination: string
  purpose?: string
  startDate: string
  endDate: string
  createdById: number
  assignedEmployeeIds: number[]
}